runtime/stringbuf.c

```diff
@@ -281,6 +281,10 @@ rsRetVal rsCStrAppendStrWithLen(cstr_t *pThis, const uchar* psz, size_t iStrLen)
 	memcpy(pThis->pBuf + pThis->iStrLen, psz, iStrLen);
 	pThis->iStrLen += iStrLen;
 
+	/* reset any cached sz string */
+	free(pThis->pszBuf);
```

Comment:
OK, take you're time, I understand if you're concerned with performance. I did handle the memleak by freeing `pszBuf` before setting it to NULL. But leaving a bug like that in the code and working around it elsewhere will probably bite you some other time and place.

I can think of other ways to handle the problem instead of freeing `pszBuf`, e.g. setting a flag that instructs `rsCStrGetSzStrNoNULL` to re-create the sz string by reallocating `pszBuf`. If you want I can implement that variant instead.